lib/cn/kblock_builder.c

```diff
@@ -1031,17 +1038,30 @@ kbb_create(struct kblock_builder **builder_out, struct cn *cn, struct perfc_set
     bld->pc = pc;
     bld->agegroup = HSE_MPOLICY_AGE_LEAF;
 
+    for (int i = HSE_MCLASS_BASE; i < HSE_MCLASS_COUNT; i++) {
```

Comment:
I think we can determine the media class for bld->agegroup + DTYPE_KEY and then query the mblock size from mpool for that media class, like how you did in kblock_finish()?

I don't know what are the consequences of passing the max mblock size of al the configured media classes here...

Reply:
For some reason I thought max_size meant the max mblock size. I will adjust.